lib/bolt/executor.rb

```diff
@@ -101,12 +101,16 @@ def shutdown
     # groups targets by transport, then divides each group into batches as
     # defined by the transport. Yields each batch, along with the corresponding
     # transport, to the block in turn and returns an array of result promises.
-    def queue_execute(targets)
-      targets.group_by(&:transport).flat_map do |protocol, protocol_targets|
+    def queue_execute(target_mapping)
+      protocol_target_mapping = target_mapping.group_by { |target, _value| target.transport }
```

Comment:
This seems like it's complicating what you really want, which is to pass around the target and associated parameters together. It's tempting to just add a `parameters` attribute to `Bolt::Target`, but we'd *really* have to make sure to clear it out after every task run. A safer option would be to have a `PlanTarget` object or something (bikeshed on name) that inherits from Bolt::Target and includes a `parameters` accessor. It would gets instantiated in `run_task_with`, and could be more easily passed around here without need to modify how other functions interact with the executor. You could maybe get away with a Struct instead of a whole class, if that seems easier.

Reply:
I think it would be almost like a `WorkRequest` with an action, parameters and a target. I'm not sure we get a whole lot out of it right now though.

Reply:
It wouldn't affect behavior, but the data we're passing around now feels sufficiently hard to express and reason about that I think it'd make the code more readable.

Reply:
Are we doing lookup on targets anywhere? At a minimum this feels more like a list of [target, params] than a hash of target -> params.